File: SignatureGenerator/Models/Signature.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Threading.Tasks;

namespace SignatureGenerator.Models
{
    [DataContract]
    public class Signature
    {
        [DataMember]
        public Dictionary<string, string> TextBlocks { get; } = new Dictionary<string, string>();
        
        [DataMember]
       public Person Person { get; set; }

        [DataMember]
        public string LegalInfo { get; set; }
    }

    [DataContract]
    public class Person
    {
        [DataMember]
        public string FirstName { get; set; }
        
        [DataMember]
        public string LastName { get; set; }
        
        [DataMember]
        public string PositionDescription { get; set; }
        
        [DataMember]
        public List<ContactData> ContactData { get; } = new List<ContactData>();
    }

    [DataContract]
    public class ContactData
    {
        [DataMember]
        public string Description { get; set; }
        [DataMember]
        public string Value { get; set; }
    }
}

File: SignatureGenerator.Shared/Model/Signature.vb
Imports System.Runtime.Serialization

<DataContract>
    Public Class Signature
        <DataMember>
        Public Property TemplateName As String
        <DataMember>
        Public ReadOnly Property TextBlocks As Dictionary(Of String, String) = New Dictionary(Of String, String)()
        <DataMember>
        Public ReadOnly Property Images As Dictionary(Of String, String) = New Dictionary(Of String, String)()
        <DataMember>
        Public Property Person As Person
        <DataMember>
        Public Property LegalInfo As String
    End Class

    <DataContract>
    Public Class Person
        <DataMember>
        Public Property FirstName As String
        <DataMember>
        Public Property LastName As String
        <DataMember>
        Public Property PositionDescription As String
        <DataMember>
        Public ReadOnly Property ContactData As List(Of ContactData) = New List(Of ContactData)()
    End Class

    <DataContract>
    Public Class ContactData
        <DataMember>
        Public Property Description As String
        <DataMember>
        Public Property Value As String
    End Class